### DISCUSSION TITLE
Schema awareness aware of keys

### DISCUSSION THREAD

**Discussion by PierreAndreis** [urql-graphql/urql#1594] [Q&A]
Can schema awareness be aware that some entities don't have an "ID" field on it and automatically ignore the error and treat as a non-keyable entities?

**Comment by kitten** [comment]
It could! 😅 the problem is that we warn about it and require a user intervention in that case because embedded entities aren't always desired; I'd classify the common cases of this warning into:

- entities that happen to have different keyable fields as a one-off
- entities that have secondary keyable fields (like email + username or the likes)
- entities that are expected to always be unique (e.g. GeoJSON data)
- sub-entities that are conceptually embedded on their parent anyway
- edges that link two entities
- pagination-related information

For edges and pagination info we do ignore the common naming schemes as standardised by Relay's spec.

In the first two cases intervention is definitely required. The third and fourth cases are hence the ones that would be automatable — but they can't be kept apart from the other two automatically
